transfers/transfer_async.py

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+"""
+Automate Transfers.
+
+Helper script to automate running transfers through Archivematica.
+
+Similar to ``transfers.transfer`` but using the new `/api/v2beta` API when
+possible.
+"""
+
+from __future__ import print_function, unicode_literals
+
+import base64
+import os
+import sys
+
+import requests
+
+# Allow execution as an executable and the script to be run at package level
+# by ensuring that it can see itself.
+sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+
+from transfers import transfer
+from transfers.loggingconfig import set_log_level
+from transfers.models import Unit
+from transfers.transferargs import get_parser
+from transfers.transfer import (
+    LOGGER, get_next_transfer, get_accession_id, main
+)
+from transfers.utils import fsencode
+
+
+class DashboardAPIError(Exception):
+    """Dashboard API error."""
+
+
+def _api_create_package(am_url, am_user, am_api_key,
+                        name, type, accession,
+                        ts_location_uuid, ts_path):
+    url = am_url + '/api/v2beta/package/'
+    headers = {
+        'Authorization': 'ApiKey {}:{}'.format(am_user, am_api_key)
+    }
+    data = {
+        'name': name,
+        'type': type,
+        'accession': accession,
+        'path': base64.b64encode(fsencode(ts_location_uuid) + b':' + ts_path),
+    }
+    LOGGER.debug('URL: %s; Headers: %s, Data: %s', url, headers, data)
+    response = requests.post(url, headers=headers, json=data)
+    response.raise_for_status()
+    LOGGER.debug('Response: %s', response)
+    resp_json = response.json()
+    error = resp_json.get('error')
+    if error:
+        raise DashboardAPIError(error)
+    return resp_json
+
+
+def _start_transfer(ss_url, ss_user, ss_api_key, ts_location_uuid, ts_path,
+                    depth, am_url, am_user, am_api_key, transfer_type,
+                    see_files, session, config_file):
+    """
```

Comment:
One thing I forgot. I wonder if there is an opportunity with this PR to consider the use of am_ss_client object to hold some of these values. There are a lot parameters to pass around. It would be nice to set  them once and early. That might open up more options, for example, enabling that object to initialise itself, e.g. from `ENV` vars in the OS. But, for now, it might be nice to just have a single object for the `am_` `ss_` parts of each new function call. 

What do you think?